File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel


class DocumentUploadResponse(BaseModel):
    success: bool
    filename: str
    category: str
    chunks_created: int


class UploadedDocument(BaseModel):
    filename: str
    category: str
    source_name: str
    chunks_created: int
    uploaded_at: datetime

File: frontend/src/types/document.ts
export type UploadedDocument = {
  filename: string;
  category: string;
  source_name: string;
  chunks_created: number;
  uploaded_at: string;
  file_size: number;
};

export type DocumentUploadPayload = {
  file: File;
  category: string;
  source_name: string;
  source_url: string;
  keywords: string;
};

export type DocumentUploadResponse = {
  success: boolean;
  filename: string;
  category: string;
  chunks_created: number;
  file_size: number;
};
